torch/_functorch/aot_autograd.py

```diff
@@ -4458,7 +4458,7 @@ def convert(idx, x):
 
         needs_autograd = (
             any(x.requires_grad for x in fake_flat_args if isinstance(x, Tensor))
-            and torch.is_grad_enabled()
+            or torch.is_grad_enabled()
```

Comment:
This is definitely a good idea. We already check later on whether any output requires_grad, and if not, fallback on `needs_autograd=False`